src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js

```diff
@@ -160,14 +160,61 @@ class ActivityFeed extends React.Component<Props, State> {
         versionInfoHandler(data);
     };
 
+    /**
+     * Determine whether or not a sort should occur, based on new comments, tasks, versions.
+     *
+     * @param {Comments} - [comments] - Object containing comments for the file.
+     * @param {Tasks} - [tasks] - Object containing tasks for the file.
+     * @param {FileVersions} - [versions] Object containing versions of the file.
+     * @return {boolean} True if the feed should be sorted with new items.
+     */
+    shouldSortFeedItems(comments?: Comments, tasks?: Tasks, versions?: FileVersions): boolean {
+        if (!comments || !tasks || !versions) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     *  If the file has changed, clear out the feed state.
+     *
+     * @param {BoxItem} [file] The box file that comments, tasks, and versions belong to.
+     * @return {Promise} - A promise that resolves in 'true' if the feedItems are empty.
+     */
+    clearFeedItems(file?: BoxItem): Promise<boolean> {
+        const { file: oldFile } = this.props;
+        if (file && file.id !== oldFile.id) {
+            return new Promise((resolve) => {
```

Comment:
Does this need to return a promise? What if you just returned a boolean? 

setState is async, but if you aren't referring to the state after its set (doesn't seem like you are), you can just assume it will get set eventually, and you can proceed with your sorting and setting the state again

Reply:
Yeah, I'm not sure what I was thinking with Promises. Fixed via boolean return